src/sentry/tsdb/snuba.py

```diff
@@ -129,7 +162,10 @@ def get_data(
         `group_on_time`: whether to add a GROUP BY clause on the 'time' field.
         `group_on_model`: whether to add a GROUP BY clause on the primary model.
         """
-        model_query_settings = self.all_model_query_settings.get(model)
+        if rollup and rollup == 10 and model in self.lower_rollup_query_settings.keys():
```

Comment:
Not sure I understand why we are going to do this only on 10 seconds rollup. Is that the only rollup value under 1 hour that we support ?

Reply:
Yea. I'll add a comment.

Reply:
Yea. I'll add a comment.

Reply:
then I think you can simplify the condition:

> if rollup == 10 and ....

But, more broadly, why not supporting any rollup lower than one hour? I think it would make sense to only support certain rollup at api level, but at backend level, I don't see a reason to force it to only works on 10 seconds.
Also, looking at the api, it does not seem it validates that condition:
https://github.com/getsentry/sentry/blob/56f5df706bc7692bc730954f6b0648280c6a33ed/src/sentry/api/base.py#L339-L343

- How does the redis TSDB behave if a user provides 20 seconds or 20 minutes as resolution ?
- did you check whether snuba and redis would align the rollup in the nsame way to the start of the time window requested by the user ?

Reply:
> How does the redis TSDB behave if a user provides 20 seconds or 20 minutes as resolution ?

Today, if you give 20 seconds or 20 minutes to the API, it'll 500. So, I feel pretty confident that we don't need to handle those cases

> did you check whether snuba and redis would align the rollup in the nsame way to the start of the time window requested by the user ?

Tried figuring this out through code but wasn't getting anywhere. So, asked @tkaemming . He says yes, RedisTSDB also aligns counts to the start of the window.

Reply:
>Today, if you give 20 seconds or 20 minutes to the API, it'll 500. So, I feel pretty confident that we don't need to handle those cases

:scream: